driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -874,7 +864,7 @@ private PooledConnection acquirePermitOrGetAvailableOpenedConnection(final boole
                 return availableConnection;
             } finally {
                 try {
-                    if (tryGetAvailable && availableConnection == null) {//the desired connection slot has not yet been removed
+                    if (expressedDesireToGetAvailableConnection && availableConnection == null) {
```

Comment:
This fixes the second problem I discovered while implementing pausable pool. `getPooledConnectionImmediately` there may throw an exception, which in turn may result in this code calling `giveUpOnTryingToGetAvailableConnection` despite `expressDesireToGetAvailableConnection` not being called.

This is an example of how one should try not to rely if possible on methods not throwing exceptions.